notebooks/example.ipynb

```diff
@@ -34,28 +34,171 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from uwtools.api import template"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9328416e-b1b8-4062-9c56-5b37440cbc4a",
+   "metadata": {},
+   "source": [
+    "We have a Jinja2 template file in `fixtures/user_config.yaml` that looks like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "67ebded6-474e-465d-81f2-ca8f76b89f54",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "user:\n",
+      "  name: {{ first }} {{ last }}\n",
+      "  favorite_food: {{ food }}\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat ./fixtures/user_config.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f29f1dc-a201-4356-a729-7b40565c8aa4",
+   "metadata": {},
+   "source": [
+    "We can use another yaml file that contains the values we want to add to the template to complete it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "5df2f19a-6cb2-4c67-8fdf-2a5aa563b4e5",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "first: John\n",
+      "last: Doe\n",
+      "food: burritos"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat ./fixtures/user_values.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd2999db-c1e7-4ae3-9169-b807b63712e7",
+   "metadata": {},
+   "source": [
+    "Using `template.render` we can render the `user_config.yaml` file using the values supplied by the `user_values.yaml` to create a complete and ready to use config file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "ce7f9265-4536-4163-9e2f-53aaa6ff0c63",
+   "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "values:\n",
-       "  date: 20240105\n",
-       "  greeting: Good Night\n",
-       "  recipient: Moon\n",
-       "  repeat: 2"
+       "'user:\\n  name: John Doe\\n  favorite_food: burritos'"
       ]
      },
-     "execution_count": 2,
+     "execution_count": 6,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "# An example Python dictionary\n",
-    "dictionary = {\"values\":{\"date\":20240105, \"greeting\":\"Good Night\", \"recipient\":\"Moon\", \"repeat\":2}}\n",
+    "# path of Jinja2 template to update\n",
+    "source = \"./fixtures/user_config.yaml\"\n",
     "\n",
-    "# Build a YAMLconfig object from the dictionary\n",
-    "config.get_yaml_config(dictionary)"
+    "# values to add\n",
+    "vals = \"./fixtures/user_values.yaml\"\n",
+    "\n",
+    "# destination of the rendered file\n",
+    "target = \"./fixtures/rendered_config.yaml\"\n",
+    "\n",
+    "# render the template \n",
+    "template.render(values_src=vals, values_format=\"yaml\", input_file=source, output_file=target)"
```

Comment:
If you wrap this line in `print`, it may look a bit nicer in the output:

```suggestion
    "print(template.render(values_src=vals, values_format=\"yaml\", input_file=source, output_file=target))"
```